pages/seacreatures.jsx

```diff
@@ -49,27 +55,109 @@ export default function SeaCreaturesPage() {
     setShowDonated(!showDonated);
   }
 
+  const sortName = () => {
+    setCrittersToShow(
+      [...crittersToShow].sort((critterA, critterB) => {
+        const nameA = critterA.name.toUpperCase();
+        const nameB = critterB.name.toUpperCase();
+        if (isAsc) {
+          if (nameA < nameB) {
+            return -1;
+          }
+          if (nameA > nameB) {
+            return 1;
+          }
+        } else {
+          if (nameA > nameB) {
+            return -1;
+          }
+          if (nameA < nameB) {
+            return 1;
+          }
+        }
+      })
+    );
+    setIsAsc(!isAsc);
+  };
+
+  const sortPrice = () => {
+    setCrittersToShow(
+      [...crittersToShow].sort((critterA, critterB) => {
+        const priceA = parseInt(critterA.price);
+        const priceB = parseInt(critterB.price);
+        if (isAsc) {
+          if (priceA < priceB) {
+            return -1;
+          }
+          if (priceA > priceB) {
+            return 1;
+          }
+        } else {
+          if (priceA > priceB) {
+            return -1;
+          }
+          if (priceA < priceB) {
+            return 1;
+          }
+        }
+      })
+    );
+    setIsAsc(!isAsc);
+  };
+
+  const selectShadows = e => {
+    const shadow = e.target.value;
+    if (shadow === 'all') {
+      return setFilter({ ...filter, shadow: null });
+    }
+
+    setFilter({ ...filter, shadow });
```

Comment:
```suggestion
  const selectShadows = event => {
    const shadow = event.target.value;

    setFilter({ ...filter, shadow: shadow === 'all' ? null : shadow });
```

You can replace the if with a ternary statement. Also remember to properly name your variables, e has no real meaning and could be interpreted as many things. In this case it's pretty obvious, but trust me, for bigger projects it's better to have all variables named properly.